zen/render-2d.c

```diff
@@ -9,21 +9,83 @@
 #include "zen/scene/screen.h"
 #include "zen/scene/view.h"
 
+typedef void (*zn_surface_iterator_func_t)(struct wlr_surface *surface,
+    struct wlr_renderer *renderer, const struct wlr_fbox *box);
+
+struct zn_surface_iterator_data {
+  zn_surface_iterator_func_t iterator;
+
+  struct zn_view *view;
+  struct wlr_renderer *renderer;
+};
```

Comment:
This may cause namespace conflicts in the future.

```suggestion
typedef void (*surface_iterator_func_t)(struct wlr_surface *surface,
    struct wlr_renderer *renderer, const struct wlr_fbox *box);

struct surface_iterator_data {
  surface_iterator_func_t iterator;

  struct zn_view *view;
  struct wlr_renderer *renderer;
};
```